fdmi/plugins/sched.c

```diff
@@ -1,31 +1,48 @@
 /* -*- C -*- */
 /*
-* Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
-*
-* Licensed under the Apache License, Version 2.0 (the "License");
-* you may not use this file except in compliance with the License.
-* You may obtain a copy of the License at
-*
-*    http://www.apache.org/licenses/LICENSE-2.0
-*
-* Unless required by applicable law or agreed to in writing, software
-* distributed under the License is distributed on an "AS IS" BASIS,
-* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-* See the License for the specific language governing permissions and
-* limitations under the License.
-*
-* For any questions about this software or licensing,
-* please email opensource@seagate.com or cortx-questions@seagate.com.
-*
-*/
+ * Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
 
 #define M0_TRACE_SUBSYSTEM M0_TRACE_SUBSYS_FDMI
 
+#include <unistd.h>
+#include <fcntl.h>
 #include "fdmi/plugins/sched.h"
-/*
+
+#define FDMI_PLUGIN_EP "/etc/fdmi_plugin_ep"
+
+static struct m0_semaphore sched_sem;
+static struct m0_config	m0_conf = {};
+static struct m0_client	*m0c = NULL;
+static struct m0_idx_dix_config	dix_conf = {};
+
+const struct m0_fdmi_pd_ops *pdo;
+static struct m0_reqh_service *sched_fdmi_service = NULL;
+
+#define MAX_LEN 8192
+static char buffer[MAX_LEN];
+
+
+#if 0
```

Comment:
This `#if 0` code seems to be used earlier for debugging, if this is not required we can remove it. We can write some comments to describe the fields and how to interpret so it can be useful for any future developers. Can change this into overview in comments here for developers?

Reply:
Please remove all unused code and make it as simple as possible.

Reply:
It can be useful for future developers, what about disabling it with #ifdef DEBUG?

Reply:
OK.
So, people know these functions are used to print debug information.

Reply:
may be move it to separate file `print_fdmi_rec.c` and include related header and minimal comment in current source. This way core sample remains small and focused on usage. What do you think?

Reply:
ok

Reply:
moved to debug.c